File: api/server/src/main/java/su/plo/voice/api/server/audio/source/ServerDirectSource.java
package su.plo.voice.api.server.audio.source;

import su.plo.voice.api.server.player.VoiceServerPlayer;
import su.plo.voice.proto.data.audio.source.DirectSourceInfo;

/**
 * @see AudioDirectSource
 */
public interface ServerDirectSource extends AudioDirectSource<VoiceServerPlayer>, ServerAudioSource<DirectSourceInfo> {
}

File: api/server-proxy-common/src/main/kotlin/su/plo/voice/api/server/audio/source/ServerDirectSource.kt
package su.plo.voice.api.server.audio.source

import su.plo.voice.api.server.player.VoicePlayer

/**
 * Represents a direct audio source attached to the player.
 *
 * Audio and source packets will be sent to this player.
 */
interface ServerDirectSource : BaseServerDirectSource {

    /**
     * Gets a player attached to this source.
     *
     * @return The voice player.
     */
    val player: VoicePlayer
}
